kitsune/sumo/utils.py

```diff
@@ -128,7 +130,16 @@ def get_next_url(request):
     else:
         url = request.META.get("HTTP_REFERER")
 
-    if not settings.DEBUG and not is_safe_url(
+    if not url:
+        return None
+
+    # Ensure that the URL is allowed as a redirect.
+    try:
+        HttpResponseRedirect(url)
```

Comment:
I am not sure that I completely understand this solution. Can you maybe elaborate on why this is working?